doc/LibcurlTransportAdapter.md

```diff
@@ -0,0 +1,162 @@
+
+### Azure SDK requirements
+
+From all the operations supported by Azure services, there's one that can't be fully completed by following the libcurl typical flow mentioned before. It is the download operation from the Azure Storage service which requires something else. When the size of the downloaded item is bigger than the amount of memory in the system, for example, downloading 1Tb from an Android device with only 8GB of memory, or from a Windows PC with 32 GB of memory, there should be no complications for an SDK client. However, based on the typical libcurl flow, during step 4, libcurl blocks the application until all the bytes from the server have been received, and that's not acceptable from the SDK perspective.
+
+#### Stream response on demand
+
+The expectation when using the SDK client to download bytes is to enable an application to read any amount of data from the network, on demand. An application must be able to read any amount of data from the network, then, it must be able to do something with that data and decide when to read some more. 
+ 
+A simple example to visualize this requirement is to think about a big download (say 1Tb) as a media file with a high quality (say a 4K movie). Watching a few seconds of the movie, from any device, should not require downloading the entire movie. Downloading a few seconds would be equivalent to reading a few hundred bytes, which are played from the device while some more seconds (data) are downloaded. Whatever is already watched can be overridden by some more content.
+
+#### No intermediate buffer
+
+Older versions of the Azure SDK have worked around the libcurl programing model by using an intermediate buffer to fetch the entire response from the server and let the program continue from step 4 (from the steps mentioned above). By doing this, the maximum download size gets limited to what the system memory can handle. Older SDK versions would support 2GB as the maximum download size.
+ 
+Another approach is to be able to request data from the server starting at some specific offset. Then, downloading big content would be equivalent to sending one request after another, asking for data starting on the last downloaded data. However, this strategy adds unnecessary extra messaging.  Each request is parsed by the server to produce a response that is sent back to the client. The client also parses the response. These server and client parsing would be happening one after each other, so it is not efficient.
+
+### Poor man's substitute
```

Comment:
What is a "Poor man's substitute"? 

Also, consider rewording this heading because it is somewhat derogatory towards poor people.